core-processor/src/executor.rs

```diff
@@ -242,51 +242,50 @@ pub fn execute_wasm<A: ProcessorExt + EnvExt + IntoExtInfo + 'static, E: Environ
     );
 
     if lazy_pages_enabled {
-        A::protect_pages_and_init_info(initial_pages, program_id, env.get_wasm_memory_begin_addr())
-            .map_err(|e| ExecutionError {
+        if let Err(e) = A::protect_pages_and_init_info(
+            initial_pages,
+            program_id,
+            env.get_wasm_memory_begin_addr(),
+        ) {
+            return Err(ExecutionError {
                 program_id,
                 gas_amount: env.drop_env(),
                 reason: e,
                 allowance_exceed: false,
-            })?;
+            });
+        }
     }
 
     // Page which is right after stack last page
     let stack_end_page = env.get_stack_mem_end();
     log::trace!("Stack end page = {:?}", stack_end_page);
 
     // Running backend.
-    let BackendReport {
-        termination,
-        wasm_memory_addr,
-        info,
-    } = match env.execute(kind.into_entry()) {
-        Ok(report) => report,
-        Err(e) => {
-            return Err(ExecutionError {
-                program_id,
-                gas_amount: e.gas_amount,
-                reason: e.reason,
-                allowance_exceed: false,
-            })
-        }
-    };
+    let BackendReport { termination, info } =
+        match env.execute(kind.into_entry(), Option::<()>::None, |wasm_memory_addr| {
```

Comment:
```suggestion
        match env.execute(kind.into_entry(), None, |wasm_memory_addr| {
```

Need to set where type for be free to use just none here. Or paste there custom enum like PreHandler::None and PreHandler::func(_)

Reply:
dicussed